src/Core/src/Hosting/Internal/MauiServiceProviderFactory.cs

```diff
@@ -28,7 +28,7 @@ public IServiceCollection CreateBuilder(IServiceCollection services)
 		public IServiceProvider CreateServiceProvider(IServiceCollection containerBuilder)
 		{
 			if (containerBuilder is IMauiServiceCollection mauiServiceCollection)
-				return new MauiServiceProvider(mauiServiceCollection, _constructorInjection);
+				return new MauiFactory(mauiServiceCollection, _constructorInjection);
```

Comment:
Sort of joking, but only a bit: Should this file/type be called `MauiFactoryFactory`?

Reply:
yeah i guess..